Remove proprietary license

neuropilot-captain opened the pull request:
### Summary
- Removed MTK proprietary header: NeauronAdapter.h from repo. Need to be downloaded from NeuroPilot Express SDK.\
- Modify license declaration in multiple header.

Fix #9279

MediaTek-ai committed:
Remove MTK proprietary files and modify header licenses

MediaTek-ai committed:
remove MTK proprietary license from llama runner

neuropilot-captain committed:
Merge branch 'main' into remove_proprietary_license

cccclai commented on backends/mediatek/runtime/include/api/NeuronAdapter.h:
```diff
@@ -1,2385 +0,0 @@
-/* Copyright Statement:
```
Is it intended to delete this file?

neuropilot-captain replied:
Yes, we have to move this file to NP Express SDK because it is a MTK proprietary header

cccclai reviewed (commented):
@mergennachin can you take a look?

neuropilot-captain commented:
@pytorchbot label "topic: not user facing"

neuropilot-captain committed:
fix lint errors

MediaTek-ai committed:
add instruction for accessing NeuronAdapter.h

MediaTek-ai committed:
remove MTK license from mtk_llama_executor_runner.cpp

neuropilot-captain commented:
Hi @mergennachin, thanks for reviewing this PR.
1. We have removed the license in the latest patch
2. 3. The users will have to copy the header to `api` directory from NeuroPilot Express SDK. We have updated the README for this. The header will be added to the SDK by this month

mergennachin commented:
Okay, I will merge this PR, as this unblocks the license issue.

But consider a few options moving, as this hinders user adoption experience:

- Automatic script that does necessary stepping the SDK, including the copying .h file.
- Provide a very thin integration header .h file that has more permissible license, and contains only the public API definitions, without any proprietary stuff that can be in any open source repo. And still have a proprietary .h file in the SDK, that defines proprietary stuff.
